lib/lorekeeper/json_logger.rb

```diff
@@ -77,10 +77,11 @@ def exception(exception, custom_message = nil, custom_data = nil, custom_level =
       if exception.is_a?(Exception)
         backtrace = clean_backtrace(exception.backtrace || [])
         exception_fields = {
+          DATA => param_data,
           EXCEPTION => "#{exception.class}: #{exception.message}",
           STACK => backtrace
         }
-        exception_fields[DATA] = param_data if param_data
+        exception_fields.compact!
```

Comment:
since you mentioned performance in the description, what about just the naive approach instead of `compact`?
```ruby
exception_fields = {}
exception_fields[DATA] = param_data if param_data
exception_fields[EXCEPTION] = "#{exception.class}: #{exception.message}"
exception_fields[STACK] = backtrace
```

Reply:
Hah, yeah, I was thinking of that or:
```
        exception_fields = {
          DATA => param_data,
          EXCEPTION => "#{exception.class}: #{exception.message}",
          STACK => backtrace
        }
        exception_fields.delete(DATA) unless param_data
```

Reply:
it's slow

```
Comparison:
              before:  3252707.9 i/s
               after:  3001680.9 i/s - same-ish: difference falls within error
             naitive:  2732396.6 i/s - same-ish: difference falls within error
```